core/src/main/java/io/questdb/griffin/SqlOptimiser.java

```diff
@@ -4799,6 +4811,23 @@ && isHourKeyword(qc.getAst().token) && qc.getAst().paramCount == 1 && qc.getAst(
         return root;
     }
 
+
+    private boolean checkIfTranslatingModelIsRedundant(boolean useInnerModel, boolean useGroupByModel, boolean useWindowModel, boolean forceTranslatingModel, boolean checkTranslatingFunction, QueryModel translatingModel) {
+        // check if translating model is redundant, e.g.
+        // that it neither chooses between tables nor renames columns
+        boolean translationIsRedundant = (useInnerModel || useGroupByModel || useWindowModel) && !forceTranslatingModel;
+        if (translationIsRedundant && !checkTranslatingFunction) {
```

Comment:
The `checkTranslatingFunction` flag is meant for checking the translating model, not a function. Also, it should be inverted:
```suggestion
        if (translationIsRedundant && checkTranslatingModel) {
```